WIP: Yi Home Cameras

bachya opened the pull request:
## Description:
Adds support for YI Home Cameras [running the alternative firmware for Xiaomi Cameras based on Hi3518e Chipset](https://github.com/shadow-1/yi-hack-v3). Awaiting some underlying library changes before pressing forward.

**Related issue (if applicable):** N/A

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** TBD

## Example entry for `configuration.yaml` (if applicable):
```yaml
camera:
  - platform: yi
    name: Kitchen Camera
    host: '192.168.1.100'
    username: root
    password: my_password_123
    ffmpeg_arguments: '-vf scale=800:450'
```

## Checklist:

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [x] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [x] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [x] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [x] New files were added to `.coveragerc`.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

bachya committed:
Playing around...

bachya committed:
Playing around...

bachya committed:
More tinkering

bachya committed:
More tinkering

bachya committed:
First "alpha" in place

bachya committed:
First "alpha" in place

bachya committed:
Removing asyncio (unnecessary)

bachya committed:
Removing asyncio (unnecessary)

bachya committed:
pylint

bachya committed:
pylint

balloob commented on homeassistant/components/camera/yi.py:
```diff
@@ -0,0 +1,144 @@
+        """Retrieve the latest video file from the customized Yi FTP server."""
+        from aioftp import ClientSession
+
+        async with ClientSession(self.host, 21, self.user,
```
We are still supporting Python 3.4 till the beginning of 2018. This means that we do not support `async` and `await` keywords. Use `yield from` / `@asyncio.coroutine`

balloob replied:
That also means that aioftp cannot be used as it requires a minimum of Python 3.5.

bachya replied:
Understood. `aioftp` offers many benefits over `ftplib` and I would prefer to not rewrite this, only to rework it again in early 2018, so I’ll hold on until Python 3.5+ becomes the standard.

bachya replied:
Understood. After considering it, I’m okay backing down to `ftplib`; will implement soon.

balloob commented on homeassistant/components/camera/yi.py:
```diff
@@ -0,0 +1,144 @@
+                                 self.passwd) as client:
+            dirs = [
+                p
+                for p, i in sorted(
```
This is unreadable, please just write it as a normal for loop

balloob committed:
Fix I/O in event loop by Arlo alarm control panel (#9738)

tchellomello committed:
Make Arlo battery_level icon dynamic (#9747)

* Make Arlo battery_level icon dynamic

* makes lint happy

snjoetw committed:
Fix for TypeError in synology camera (#9754)

rytilahti committed:
switch.tplink, light.tplink: bump the pyhs100 version and adapt to api changes (#9454)

* bump the pyhs100 version and fix api changes

* switch.tplink: avoid I/O during __init__

* initialize _name to None in __init__

* update requirements_all.txt for the new version

balloob committed:
RFC: Use bind_hass for helpers (#9745)

* Add Helpers bind_hass functionality

* Update other helpers

bachya committed:
Rewrite using ftplib

bachya committed:
Rewrite using ftplib

bachya commented:
@balloob: I've rewritten this using `ftplib`. One area of concern I have has to do with image retrieval. Since I'm a relative newbie to async programming, I'd appreciate your input.

I'm using @pvizeli's `[ha-ffmpeg](https://github.com/pvizeli/ha-ffmpeg)` library to process the MP4, which, I believe, necessitates using the `async_camera_image` method (instead of `camera_image`). Unless I'm mistaken, doesn't line 122 represent IO going on inside the event loop?

I'm using this as a custom component in my HASS install and everything appears fine (camera is updating, no timer desynchronization, etc.), but would love to be sure.

bachya committed:
Fixing linter issues

bachya committed:
Fixing linter issues

bachya committed:
Missing @asyncio.coroutine

bachya committed:
Missing @asyncio.coroutine

bachya committed:
Added try/except for cases where video folder is empty

bachya committed:
Added try/except for cases where video folder is empty

bachya committed:
Empty folder should be info, not warning

bachya committed:
Empty folder should be info, not warning

bachya committed:
Attempting fix of image blocking IO

bachya committed:
Attempting fix of image blocking IO

MisterWil committed:
Skybell (#9681)

* New Skybell platform with components

* Added skybell components to omit.

* Preemptively fixing lint issues (hopefully).

* Removed unused variable.

* Requested changes.

* Additional CRs

* Hopefully the last of the CR's!

rytilahti committed:
add myself to yeelight owners, too (#9759)

bachya committed:
Updated CODEOWNERS

Danielhiversen committed:
Update CODEOWNERS (#9760)

bachya committed:
Adding my contributions (#9761)

ChristianKuehnel committed:
Initializing statistics sensor with data from database (#9753)

* Initializing statistics sensor with data from database

* fixed broken test case

* usage of recorder component is now optional, thx to @andrey-git

* added test case for initialize_from_database

andrey-git committed:
Match test requirements by full package name. (#9764)

rytilahti committed:
yeelight: implement min_mireds and max_mireds, fixes #9509 (#9763)

* yeelight: implement min_mireds and max_mireds, fixes #9509

thanks to @amelchio for pointing this out!

* remove typing infos

tchellomello committed:
Bump raincloudy version 0.0.3 (#9767)

* Bump raincloudy version 0.0.3

* Fix logic for raincloudy status binary_sensor

* Simplified binary_sensor logic

* Simplify

balloob commented on homeassistant/components/camera/ffmpeg.py:
```diff
@@ -6,7 +6,6 @@
 """
 import asyncio
 import logging
-
```
Revert this change

balloob commented on homeassistant/components/camera/yi.py:
```diff
@@ -0,0 +1,143 @@
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_NAME):
+    cv.string,
```
Please keep the key and value on one line if it fits.

bachya replied:
Will do. FYI, [yapf](https://github.com/google/yapf) is doing this.

syssi committed:
Xiaomi Smart WiFi Socket and Smart Power Strip integration (#9138)

* Xiaomi Smart WiFi Socket and Smart Power Strip integration

* Comment updated.

* Blank line removed.

* Typo fixed.

* Version of python-mirobo bumped.

* Version of python-mirobo bumped: Lightweight API changes.

* Additional API changes.

* Library version properly pinned again.

* Platform not ready behavior fixed.
Expose the device model as sensor attribute.
Device initialized log message added. Provides device model, firmware and hardware version.

* Component renamed: switch.xiaomi_plug -> switch.xiaomi_miio

* Revise based on review: Unused code removed. Filename updated.

etsinko committed:
fix for LocationParseError in netgear platform (#9683)

* fix for LocationParseError in netgear platform

* added unit tests for get_scanner()

* fixed houndci-bot warnings

* fixed lint warnings

* fixed lint warnings

* fixed broken test

* removed guard clause from netgear.py
removed all discovery related code from device_tracker
removed unnecessary unit test

* removed discovery related tests

* removed unused import

* removed unused import

balloob committed:
Expose time module in Python Scripts (#9736)

* Expose time module in Python Scripts

* Make dt_util available in Python Scripts

* Limit methods in time module

* Add time.mktime

* Limit access to datetime

* Add warning to time.sleep

* Lint

webworxshop committed:
Add notification platform for Rocket.Chat. (#9553)

* Add notification platform for Rocket.Chat.

* Changes to Rocket.Chat notification platform based on feedback.

* Implement better error handling for Rocket.Chat platform.

* Return None if Rocket.Chat notify platform init fails.

* Refactor Rocket.Chat notifications.

Refactor Rocket.Chat notification platform to remove async and
simplify error handling.

* fix url

jeroenterheerdt committed:
Updating pythonegardia package requirement to .22 because of fixed bug in passing default value for parameter SSL for egardiaserver (#9770)

jeroenterheerdt committed:
Adding myself as codeowner for egardia alarm control panel. (#9772)

Adding jeroenterheerdt as codeowner for egardia alarm control panel.

bachya committed:
WIP: Fix Arlo Camera blocking IO (#9758)

* WIP: Fix Arlo Camera blocking IO

* Accidental undo

* Linting issues

* Owner-requested changes

* Bumped pyarlo version and added Throttle

* Fix

* Update requirements_all.txt

zabuldon committed:
Tesla bug fixes. (#9774)

* Tesla bug fixes.

* Added myself to CODEOWNERS for tesla.

sdague committed:
Fix off by one error in arwn platform (#9781)

There is an off by one error that causes period exceptions. Fix this.

Commit:
missing is_closed ( rflink cover fix ) (#9776)

* Added is_closed

* whitespaces --

* removed whitespace

Commit:
Wink dome siren support (#9667)

* Support for Wink Dome siren/chimes

sdague committed:
Bump rxv library to 0.5.1 (#9784)

This fixes some bugs with interfacing with yamaha receivers, including
closing bug #5209.

bachya committed:
Rebasing

bachya committed:
Owner-requested changes + merge

bachya commented:
Whoa – not sure what happened with that rebase. Going to close this out and submit a new PR (with the latest code).
